hoomd/hpmc/ExternalFieldLattice.h

```diff
@@ -348,237 +360,74 @@ template<class Shape> class ExternalFieldLattice : public ExternalFieldMono<Shap
+    //! Calculate the energy associated with the deviation of a single particle from its reference position
+    Scalar calcE_trans(const unsigned int& index, const vec3<Scalar>& position, const Scalar& scale = 1.0)
         {
         ArrayHandle<unsigned int> h_tags(m_pdata->getTags(),
                                          access_location::host,
                                          access_mode::read);
         int3 dummy = make_int3(0, 0, 0);
         vec3<Scalar> origin(m_pdata->getOrigin());
         const BoxDim& box = this->m_pdata->getGlobalBox();
-        vec3<Scalar> r0(m_latticePositions.getReference(h_tags.data[index]));
+        vec3<Scalar> r0 = m_lattice_positions[h_tags.data[index]];
         r0 *= scale;
         Scalar3 t = vec_to_scalar3(position - origin);
         box.wrap(t, dummy);
         vec3<Scalar> shifted_pos(t);
         vec3<Scalar> dr = vec3<Scalar>(box.minImage(vec_to_scalar3(r0 - position + origin)));
-        return m_k * dot(dr, dr);
+        return m_k_translational * dot(dr, dr);
         }
 
+    //! Calculate the energy associated with the deviation of a single particle from its reference orientation
     Scalar calcE_rot(const unsigned int& index, const quat<Scalar>& orientation)
         {
         assert(m_symmetry.size());
         ArrayHandle<unsigned int> h_tags(m_pdata->getTags(),
                                          access_location::host,
                                          access_mode::read);
-        quat<Scalar> q0(m_latticeOrientations.getReference(h_tags.data[index]));
+        quat<Scalar> q0 = m_lattice_orientations[h_tags.data[index]];
         Scalar dqmin = 0.0;
         for (size_t i = 0; i < m_symmetry.size(); i++)
             {
             quat<Scalar> equiv_orientation = orientation * m_symmetry[i];
             quat<Scalar> dq = q0 - equiv_orientation;
             dqmin = (i == 0) ? norm2(dq) : fmin(dqmin, norm2(dq));
             }
-        return m_q * dqmin;
+        return m_k_rotational * dqmin;
         }
+
     Scalar calcE_rot(const unsigned int& index, const Shape& shape)
         {
         if (!shape.hasOrientation())
             return Scalar(0.0);
 
         return calcE_rot(index, shape.orientation);
         }
+
+    /** Calculate the total energy associated with the deviation of a single particle from its ref. pos. and orientation
+     *
+     * This function _should_ only be used for logging purposes and not for calculating move acceptance criteria, since
+     * it's the energy difference that matters for the latter.
+     */
     Scalar calcE(const unsigned int& index,
                  const vec3<Scalar>& position,
                  const quat<Scalar>& orientation,
                  const Scalar& scale = 1.0)
         {
         Scalar energy = 0.0;
-        if (m_latticePositions.isValid())
-            {
+        //if (m_lattice_positions.isValid())
```

Comment:
This commented out code will leave once I figure out the the particle change number signal bit.

Reply:
I figured that other bit out. This is gone now.